core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -114,6 +116,13 @@ class AdminClient(val time: Time,
     listAllGroupsFlattened.filter(_.protocolType == ConsumerProtocol.PROTOCOL_TYPE)
   }
 
+  def listGroupOffsets(groupId: String): Map[TopicPartition, PartitionData] = {
+    val coordinator = findCoordinator(groupId)
+    val responseBody = send(coordinator, ApiKeys.OFFSET_FETCH, OffsetFetchRequest.allPartitionsOffsetFetchRequest(groupId))
+    val response = responseBody.asInstanceOf[OffsetFetchResponse]
+    response.responseData().asScala.toMap
```

Comment:
The basic issue is that the error codes are in the partition data of the response schema. If we have no partitions to return, then we cannot return any errors either. This is fine for most cases because we should already know if the group exists or not. However, there are (at least) two problematic edge cases:

1. The case I mentioned above. To use the OffsetFetch API, we must first lookup the coordinator for the group. It could happen that when we do so, we happen to get a stale coordinator. This is possible because it takes some time for metadata changes to propagate to all the brokers. 
2. When the coordinator first is started, it must read through the `__consumer_offsets` topic to populate the offset cache. Usually we return a `COORDINATOR_NOT_AVAILABLE` error in this situation which lets the consumer know it needs to retry a bit later. 

So if we happen to send an offset fetch for all partitions in either of these cases, then we could mistakenly believe that there are no offsets for the group.